pkg/volume/csi/csi_plugin.go

```diff
@@ -435,11 +436,23 @@ func (p *csiPlugin) NewMounter(
 	attachID := getAttachmentName(volumeHandle, driverName, node)
 	volData[volDataKey.attachmentID] = attachID
 
-	if err := saveVolumeData(dataDir, volDataFileName, volData); err != nil {
-		if removeErr := os.RemoveAll(dataDir); removeErr != nil {
-			klog.Error(log("failed to remove dir after error [%s]: %v", dataDir, removeErr))
+	err = saveVolumeData(dataDir, volDataFileName, volData)
```

Comment:
should we do same thing in this file for other places where `saveVolumeData` is being called followed up by `os.RemoveAll`? This applies to `NewBlockVolumeMapper` call.

Reply:
Is this directory same or different than the one used in MountDevice?

Reply:
As best I can tell from testing with the hostpath driver, these appear to be different directories:

* csi_plugin: created path successfully [/var/lib/kubelet/pods/103bb9c7-0a9f-4958-bb6a-8d05f64420f5/volumes/kubernetes.io~csi/pvc-21228cc8-5669-4c0d-bbdc-d7b4e19dfb79]
* csi_attacher: created target path successfully [/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-22c63cbf-4e2c-4ca6-b269-d5e44a1f3718/globalmount]